Cognite.Simulator.Utils/SimulationScheduler.cs

```diff
@@ -148,70 +165,86 @@ await Task.Run(async () =>
                             continue;
                         }
 
-                        // Check if the job already exists and if it should be cancelled due 
-                        // to a new configuration on the API
-                        if (scheduledJobs.TryGetValue(routineRev.RoutineExternalId, out var job))
+                        // Parse the cron schedule.
+                        CrontabSchedule schedule;
+                        try
                         {
-                            if (routineRev.CreatedTime > job.CreatedTime && job.TokenSource != null && !job.TokenSource.Token.IsCancellationRequested)
-                            {
-                                _logger.LogDebug($"Cancelling job for RoutineRevision : {routineRev.ExternalId} due to new configuration detected.");
-                                job.TokenSource.Cancel();
-                                scheduledJobs.Remove(routineRev.RoutineExternalId);
-                            }
+                            schedule = ParseCronTabSchedule(routineRev.Configuration.Schedule.CronExpression, _logger);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogError($"Error parsing cron expression {routineRev.Configuration.Schedule.CronExpression} " +
+                                $"for routine: {routineRev.RoutineExternalId}, error: {ex.Message}");
+                            continue;
+                        }
+
+                        // Get the last time this routine was triggered.
+                        DateTime lastRun;
+                        if (!lastRunTimes.TryGetValue(routineRev.RoutineExternalId, out lastRun))
+                        {
+                            lastRun = DateTime.MinValue;
                         }
 
-                        if (!scheduledJobs.TryGetValue(routineRev.RoutineExternalId, out var existingJob))
+                        // Compute the next scheduled occurrence from the last run time.
+                        var nextOccurrence = GetNextOccurrence(schedule, lastRun);
+
+                        // If the next occurrence is less or equal than the current UTC time (plus tolerance)
+                        // and is greater than the last run time, trigger the simulation run.
+                        if (nextOccurrence <= DateTime.UtcNow.Add(tolerance) && nextOccurrence > lastRun)
```

Comment:
```suggestion
                        if (nextOccurrence >= DateTime.UtcNow && nextOccurrence <= DateTime.UtcNow.Add(tolerance) && nextOccurrence > lastRun)
```
otherwise it might fire before the actual schedule (we can't allow that)